contracts/GenArt721CoreV3.sol

```diff
@@ -631,16 +627,24 @@ contract GenArt721CoreV3 is ERC721Enumerable, IGenArt721CoreContractV3 {
             uint256 royaltyFeeByID
         )
     {
-        artistAddress = projectIdToArtistAddress[tokenIdToProjectId[_tokenId]];
-        additionalPayee = projectIdToAdditionalPayee[
-            tokenIdToProjectId[_tokenId]
-        ];
+        uint256 projectId = _tokenId / ONE_MILLION;
+        artistAddress = projectIdToArtistAddress[projectId];
+        additionalPayee = projectIdToAdditionalPayee[projectId];
         additionalPayeePercentage = projectIdToAdditionalPayeePercentage[
-            tokenIdToProjectId[_tokenId]
-        ];
-        royaltyFeeByID = projectIdToSecondaryMarketRoyaltyPercentage[
-            tokenIdToProjectId[_tokenId]
+            projectId
         ];
+        royaltyFeeByID = projectIdToSecondaryMarketRoyaltyPercentage[projectId];
+    }
+    
+    /**
+     * @notice Gets the project ID for a given `_tokenId`.
+     */
+    function tokenIdToProjectId(uint256 _tokenId) 
+        public
+        view
+        returns (uint256 _projectId)
+    {
+        return _tokenId / ONE_MILLION
     }
```

Comment:
Given how simple this logic is, rather than actually call this `tokenIdToProjectId` method through this contract where token->project ID mapping is needed, I'm just doing `_tokenId / ONE_MILLION` inline to avoid extra gas associated w/ internal call vs. just the DIV operation if that rationale tracks from your perspective @ryley-o

Reply:
Note that `tokenIdToProjectId` is still implemented as it is part of the public interface, which I think probably makes sense as I think providing this as a public view method is helpful for integrators/folks reading the contract who may not already know the `_tokenId / ONE_MILLION` assumption without seeing this–though we could strip this method as well.

Reply:
I personally like keeping `tokenIdToProjectId` in the public interface, but agree with not using it internally as you describe, but it's a pretty core assumption to our contract and I think it's good to 'formalize' that on chain

Reply:
yup, good to not add storage costs to mint sequence storing deterministic information, but instead keep the interface avail through a new public function.

Reply:
Roger that :) glad this approach SG. Resolving.

Reply:
oh I see what you are saying about inline elsewhere. yes that tracks. The cheapest way to do this will always be inline `_tokenId / ONE_MILLION`.
nit: can you make this external and pure (I think we can do pure since ONE_MILLION is a constant)

Reply:
Yessser. Pushing a commit with those `external`/`pure` changes.